lib/sass_spec/stats.rb

```diff
@@ -0,0 +1,146 @@
+# encoding: utf-8
+
+require 'command_line_reporter'
+require 'pathname'
+require 'yaml'
+
+class SassSpec::Stats::CLI
+  include CommandLineReporter
+
+  def self.parse(args)
+    options = {}
+    parser = OptionParser.new
+    parser.banner = <<BANNER
+Usage: ./stats.rb [options] [directories...]
+
+This command prints stats about the repo.
+
+BANNER
+
+    parser.on("--todo IMPLEMENTATION",
+              "Find specs that have a todo for the given implementation.") do |impl|
+      options[:todo] = impl
+    end
+
+    parser.on("--issue GITHUB_ISSUE",
+              "Find specs that correspond to a Github issue.") do |issue|
+      unless issue =~ /.+#\d+/
+        warn("Invalid issue.\n\n")
+        puts parser.help()
+      end
+      issue = issue.split("#")
+      number = issue.pop
+      issue = issue.join().split("/")
+      impl = issue.pop
+      options[:issue] = "#{impl}##{number}"
+    end
+
+    parser.on("--missing-issue",
+              "Find specs that have a todo without a Github issue.") do
+      options[:missing_issue] = true
+    end
+
+    parser.on("-h", "--help", "Print this help message.") do
+      puts parser.help()
+      return nil
+    end
+
+    parser.parse!(args)
+    options[:dirs] = args.dup.map{ |d| File.expand_path(d) } if args.any?
+    new(options)
+  end
+
+  def initialize(options)
+    @options = options
+  end
+
+  ##
+  # Given the options provided to the CLI, filters through spec metadata to
+  # aggregate statistics about the specs. Prints a stats report.
+  def stats
+    metadata = get_metadata()
+
+    if @options[:todo]
+      metadata = filter_todos_by_implementation(metadata)
+    end
```

Comment:
Style nit:

```suggestion
    metadata = filter_todos_by_implementation(metadata) if @options[:todo]
```

This will let you write what currently takes 11 lines in 3 without losing much readability :smiley:.